solarforecastarbiter/metrics/deterministic.py

```diff
@@ -209,3 +213,101 @@ def centered_root_mean_square(y_true, y_pred):
     return np.sqrt(np.mean(
         ((y_pred - np.mean(y_pred)) - (y_true - np.mean(y_true))) ** 2
     ))
+
+
+def kolmogorov_smirnov_integral(y_true, y_pred, normed=False):
+    """Kolmogorov-Smirnov Test Integral (KSI).
+
+    Parameters
+    ----------
+    y_true : array-like
+        True values.
+    y_pred : array-like
+        Predicted values.
+    normed : bool, optional
+        If True, return the normalized KSI [%].
+
+    Returns
+    -------
+    ksi : float
+        The KSI between the true and predicted values.
+
+    """
+
+    # empirical CDF
+    ecdf_obs = ECDF(y_true)
+    ecdf_fx = ECDF(y_pred)
+
+    # evaluate CDFs on forecast range
+    x = np.linspace(np.min(y_pred), np.max(y_pred), 100)
```

Comment:
`x = np.sort(np.concatenate((y_true, y_pred))` avoids the interpolation I think.  ECDF will extrapolate on both sides. We should be explicit in the documentation that we're using a right endpoint rule (default for ECDF), e.g., if f = ECDF([1.0, 2.0]) then f(x) = 0.5 for any x <= 1.0.

Reply:
Revised per suggestion.